javascripts/discourse/initializers/initialize-for-header-icon-links.gjs

```diff
@@ -1,22 +1,16 @@
 import { withPluginApi } from "discourse/lib/plugin-api";
-import { iconNode } from "discourse-common/lib/icon-library";
+import { iconHTML } from "discourse-common/lib/icon-library";
 import { dasherize } from "@ember/string";
 import isValidUrl from "../lib/isValidUrl";
-import { h } from "virtual-dom";
+import { htmlSafe } from "@ember/template";
 
-function buildIcon(icon) {
-  if (isValidUrl(icon)) {
-    return h(
-      "img",
-      {
-        attributes: {
-          src: icon,
-        },
-      },
-      ""
-    );
+function buildIcon(iconNameOrImageUrl) {
+  if (isValidUrl(iconNameOrImageUrl)) {
+    return <template>
+      <img src="{{iconNameOrImageUrl}}"/>
```

Comment:
(It looks like we weren't adding alt text before—should we change that?)

Reply:
It looks like the `iconHTML` (and `d-icon` helper) both add `aria-hidden='true'` to the svg, so perhaps that's the solution here as well. The wrapping `<a>` already has a `title` which I assume is sufficient for a screen reader.

Reply:
🤔 My understanding is that `title` is intended for hover tooltips, while `alt` is meant for screen readers and when the image doesn't load. But I could be off here!

I've tried out the `d-icon` helper and it looks like I can pass a label in the options, which is rendered as a `sr-only` span. Should I put the title there, or should we just assume that the current title treatment is fine? I think the handlebars helper does not support that options object, so instead it would be like:
```
    const renderedIcon = icon(iconNameOrImageUrl, {label: title})
    return <template>{{renderedIcon}}</template>
```

Reply:
Ah nice, the sr-only span sounds like a good solution 👌 

I *think* you should be able to pass options to the helper directly in hbs by doing

```gjs
<template>{{icon iconNameOrImageUrl label=title}}</template>
```

Here's [one random example](https://github.com/discourse/discourse/blob/ec3d29a1fa7e57c1a18e5b1c292ee718a0193442/app/assets/javascripts/discourse/app/components/form-template-field/upload.gjs#L91-L91) of icon options in Discourse core.

Reply:
Aha! Nice, thanks 😄 

I went ahead and added a similar span in the case that we render an image instead of using an icon. I was unsure whether to use an `alt` on the image tag, but I opted for more consistent behavior from the `buildIcon` function.